web/html/src/manager/content-management/shared/components/panels/sources/channels/channels-api.ts

```diff
@@ -0,0 +1,198 @@
+) => {
+  // Create a two-way mapping of what channels require what channels, and what channels are required by what channels
+  const requiresMap = new Map<number, Set<BaseChannelType | ChildChannelType> | undefined>();
+  const requiredByMap = new Map<number, Set<BaseChannelType | ChildChannelType> | undefined>();
+  for (const channelIdString in rawRequiresMap) {
+    const channelId = parseInt(channelIdString, 10);
+    const channel = channelsMap.get(channelId);
+    if (isNaN(channelId) || !channel) {
+      throw new RangeError("Invalid channel id");
+    }
+
+    const requiredChannels = rawRequiresMap[channelIdString]
+      ?.map((id) => channelsMap.get(id) as BaseChannelType | ChildChannelType) // We know these values exist
+      .filter((requiredChannel) => requiredChannel.id !== channel.id); // The original data includes the channel's own id, don't include it in the set
+    if (requiredChannels?.length) {
+      requiresMap.set(channelId, new Set(requiredChannels));
+      requiredChannels.forEach((requiredChannel) => {
+        if (requiredByMap.has(requiredChannel.id)) {
+          requiredByMap.get(requiredChannel.id)?.add(channel);
+        } else {
+          requiredByMap.set(requiredChannel.id, new Set([channel]));
+        }
+      });
+    }
+  }
+  return { requiresMap, requiredByMap };
+};
+
+const useChannelsApi = () => {
+  const [channelsPromise, setChannelsPromise] = useState<Promise<ChannelTreeType[]> | undefined>(undefined);
+  if (channelsPromise) {
+    return [channelsPromise];
+  }
+
+  const promise = Network.get<JsonResult<ChannelTreeType[]>>(`/rhn/manager/api/channels?filterClm=true`)
+    .then(Network.unwrap)
+    .then((channels) => {
+      // TODO: Only for testing
+      if (false) {
+        for (var ii = 0; ii < MOCK_COUNT; ii++) {
+          const id = MOCK_ID_START + ii;
+          const base = {
+            id,
+            name: `mock channel ${ii} ${"filler ".repeat(100)}`,
+            label: `mock_channel_${ii}`,
+            archLabel: "channel-x86_64",
+            custom: true,
+            isCloned: false,
+            subscribable: true,
+            recommended: false,
+            standardizedName: "",
+            recommendedChildren: [],
+          };
+          channels.push({
+            base,
+            children: [
+              {
+                id: id * 2,
+                name: `mock channel child ${ii} ${"filler ".repeat(100)}`,
+                label: `mock_channel_child_${ii}`,
+                archLabel: "channel-x86_64",
+                custom: true,
+                isCloned: false,
+                subscribable: true,
+                recommended: false,
+                standardizedName: "",
+                parent: base,
+              },
+            ],
+          });
+        }
+      }
+
+      // TODO: This matches the current behavior but this should already be done on the server side, with a separate flag if necessary
```

Comment:
@cbbayburt Is this something we can move to the server, or is there a reason why not to do it that I'm missing?

Reply:
I've made the change in[`ChannelsApiController.java`](https://github.com/uyuni-project/uyuni/pull/4739/files#diff-4d83abbd4ca969f44973a005de0adc5886dd4772d704886ab010eff4008c8280), if you have a moment to check that my copy-paste-from-google Java coding makes sense, I'd appreciate it. 😄